src/main/java/ch/njol/skript/lang/Variable.java

```diff
@@ -590,53 +592,45 @@ public void change(Event e, @Nullable Object[] delta, ChangeMode mode) throws Un
 						}
 					}
 				} else {
-					Object o = get(e);
-					ClassInfo<?> ci;
-					if (o == null) {
-						ci = null;
-					} else {
-						Class<?> c = o.getClass();
-						assert c != null;
-						ci = Classes.getSuperClassInfo(c);
-					}
-					Arithmetic a = null;
+					Object object = get(e);
+					Class<?> clazz = object == null ? null : object.getClass();
```

Comment:
The newly update singleton line doesn't take into consideration `Classes.getSuperClassInfo`

You're removing ClassInfo checking for your isAssignable in Arithmetics.

It should be checking against ClassInfos.

Reply:
Why? It works fine